### ISSUE TITLE
Refactor abstract base-service and the different implementations

### ISSUE THREAD

**Issue by oysteoh** [equinor/ert#2971]
The current implementation of base-service is massive and not very easy to get a grasp of. It is error-prone, and it is hard to figure out what is causing our issues. We should possible look into if there is other ways of doing what we are trying to achieve with spawning the different services in ERT and make some proper architecture around this.

**Comment by jondequinor** [comment]
Have we ruled out systemd for our use case? It was made to start/stop/monitor services both in the system and user sessions.

**Comment by oysteoh** [comment]
From my understanding nothing is ruled out or even really discussed. I just created this issue to have it and discuss it at some point, because now when we are working on this and testing a lot - we just see that it is unstable, we struggle a lot with it and it is hard to understand what is wrong. It brings a lot of complexity orchestrating the different services. Had it worked perfectly it would have been nice, but it seems to be part of what is flaky and it is really hard to debug.   If you have suggestions please feel free :)

**Comment by jondequinor** [comment]
We can rule out `systemd` as an option as it has no support in RHEL7.

**Comment by jondequinor** [comment]
I will take a look at service managers like http://supervisord.org

**Comment by jondequinor** [comment]
Yeah, I now recall having worked extensively with supervisor in a previous project. It works well for non-privileged process control systems.

**Comment by oyvindeide** [comment]
I will close this issue as stale.
